kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/steps/KafkaSteps.java

```diff
@@ -38,6 +47,36 @@ public static void createTopic(String clusterName, String topicName, String name
                 KafkaTopicTemplates.defaultTopic(namespace, clusterName, topicName, partitions, replicas, minIsr).build());
     }
 
+    /**
+     * Create topic using test clients.
+     *
+     * @param topicName the topic name
+     * @param bootstrap the bootstrap
+     * @param partitions the partitions
+     * @param replicas the replicas
+     */
+    public static void createTopicTestClient(String deployNamespace, String topicName, String bootstrap, int partitions, int replicas) {
+        String podName = KafkaUtils.adminTestClient(deployNamespace, bootstrap);
+        String command = "admin-client topic create --bootstrap-server=" + bootstrap + " --topic=" + topicName + " --topic-partitions=" + partitions +
+                " --topic-rep-factor=" + replicas;
+
+        ByteArrayOutputStream baosOut = new ByteArrayOutputStream();
+        ByteArrayOutputStream baosErr = new ByteArrayOutputStream();
+        var exitCode = kubeClient().getClient().pods().inNamespace(deployNamespace).withName(podName)
+                .writingOutput(baosOut)
+                .writingError(baosErr)
+                .exec("sh", "-c", command)
+                .exitCode().join();
+
+        if (exitCode != 0) {
+            LOGGER.error(baosErr.toString());
```

Comment:
Two things.

- in general, never log a string directly (i.e. `LOGGER.error(myString)`) unless you can be 100% certain that the string won't contain parameterised logging tokens (`{}` etc).   Otherwise you could get misleading results.
- it is more efficient to let the logging library stringify the arguments.     At the moment with your code `toString()` is executed unconditionally, even if logging level config means error is turned off.   

```suggestion
            LOGGER.error("Execution failed: {}", baosErr);
```

Reply:
Agree. Thanks!